autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/persister/CredentialsMergeStrategy.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.sync.persister
+
+import com.duckduckgo.autofill.sync.CrendentialsSyncEntries
+import com.duckduckgo.sync.api.engine.SyncMergeResult
+
+interface CredentialsMergeStrategy {
+    fun processEntries(credentials: CrendentialsSyncEntries): SyncMergeResult<Boolean>
```

Comment:
The API gives no indication that it will be expensive to run or block the calling thread internally. We should at least mark this with a worker thread annotation?

Reply:
this is now annotated thanks.